doc/modules/neighbors.rst

```diff
@@ -510,6 +510,67 @@ the model from 0.81 to 0.82.
     classification using nearest centroid with different shrink thresholds.
 
 
+Nearest Neighbors Transformer
+=============================
+
+Many scikit-learn estimators rely on nearest neighbors: Several classifiers and
+regressors such as :class:`KNeighborsClassifier` and
+:class:`KNeighborsRegressor`, but also some clustering methods such as
+:class:`~sklearn.cluster.DBSCAN` and
+:class:`~sklearn.cluster.SpectralClustering`, and some manifold embeddings such
+as :class:`~sklearn.manifold.TSNE` and :class:`~sklearn.manifold.Isomap`.
+
+All these estimators can compute internally the nearest neighbors, but most of
+them also accept precomputed nearest neighbors :term:`sparse graph`,
+as given by :func:`~sklearn.neighbors.kneighbors_graph` and
+:func:`~sklearn.neighbors.radius_neighbors_graph`. With mode
+`mode='connectivity'`, these functions return an binary adjacency sparse graph
+as required for instance in :class:`~sklearn.cluster.SpectralClustering`,
+whereas with `mode='distance'`, they return a distance sparse graph as required
+for instance in :class:`~sklearn.cluster.DBSCAN`. The benefits of precomputation
+are multiple.
+
+First, the precomputed graph can be re-used multiple times, for instance while
+varying a parameter of the estimator. This can be done manually, or using the
+caching properties of the scikit-learn pipeline, with the help of
+:class:`KNeighborsTransformer` and :class:`RadiusNeighborsTransformer` to
+include the precomputation in a pipeline:
+
+    >>> from sklearn.manifold import Isomap
+    >>> from sklearn.neighbors import KNeighborsTransformer
+    >>> from sklearn.pipeline import make_pipeline
+    >>> estimator = make_pipeline(
+    ...     KNeighborsTransformer(n_neighbors=5, mode='distance'),
+    ...     Isomap(neighbors_algorithm='precomputed'),
+    ...     memory='/path/to/cache')
+
+Second, these transformers give finer control on the nearest neighbors
```

Comment:
Is this a benefit of precomputing or the transformers?

Reply:
I tried to reformulate.